aineko/core/dataset.py

```diff
@@ -1,529 +1,213 @@
 
-        Returns:
-            next or last value in self.values
+    ```
+    - `_read`
+    - `_write`
+    - `_create`
+    - `_delete`
+    - `_initialize`
+    - `_exists`
+    ```
 
-        Raises:
-            ValueError: if how is not "next" or "last"
-        """
-        if how not in ["next", "last"]:
-            raise ValueError(f"Invalid how: {how}. Expected 'next' or 'last'.")
-
-        if how == "next":
-            remaining = len(self.values)
-            if remaining > 0:
-                if remaining == 1:
-                    self.empty = True
-                return {
-                    "timestamp": datetime.datetime.now().strftime(
-                        AINEKO_CONFIG.get("MSG_TIMESTAMP_FORMAT")
-                    ),
-                    "message": self.values.pop(0),
-                    "source_node": "test",
-                    "source_pipeline": "test",
-                }
-        if how == "last":
-            if self.values:
-                return self.values[-1]
-
-        return None
-
-    def next(self) -> Optional[dict]:
-        """Wraps `consume(how="next")`, blocks until available.
+    Example:
+    ```python
+    class MyDataset(AbstractDataset):
+        def _read(self, **kwargs) -> Any:
+            pass
 
-        Returns:
-            msg: message from the dataset
-        """
-        return self.consume(how="next")
+        def _write(self, **kwargs) -> Any:
+            pass
 
-    def last(self, timeout: float = 1) -> Optional[dict]:
-        """Wraps `consume(how="last")`, blocks until available.
+        def _create(self, **kwargs) -> Any:
+            pass
 
-        Returns:
-            msg: message from the dataset
-        """
-        return self.consume(how="last", timeout=timeout)
+        def _delete(self, **kwargs) -> Any:
+            pass
 
+        def _initialize(self, **kwargs) -> Any:
+            pass
 
-class FakeDatasetProducer:
-    """Fake dataset producer for testing purposes.
+        def _exists(self, **kwargs) -> bool:
+            pass
+    ```
 
-    The class will store in its state the list of values produced
-    by the node.
+    If `MyDataset` was defined in the file
+    `./aineko/datasets/mydataset.py`, a new dataset
+    can be created using the `from_config` method:
 
-    Args:
-        dataset_name: name of the dataset
-        node_name: name of the node that is producing the dataset
+    Example:
+    ```python
 
-    Attributes:
-        dataset_name (str): name of the dataset
-        node_name (str): name of the node that is producing the dataset
-        values (list): list of mock data values produced by the node
+    dataset = AbstractDataset.from_config(
+        name="my_dataset_instance",
+        config={
+            "type": "aineko.datasets.mydataset.MyDataset",
+            "location": "foo",
+            "params": {
+                "param_1": "bar"
+            }
+        }
+    )
+    ```
     """
```

Comment:
I just noticed that the docstring still mentions the method names with an underscore. This should be updated before merging please.

Reply:
Updated it in  dc6552b98fc52a4cff6c3fcb05e7b8cab24c7210